Azure-Samples/modern-data-warehouse-dataops
# feat: fix ADF and CD pipelines, fix miscellaneous problems

# Type of PR
The PR will address several issues existing in the sample

- Documentation changes
- Code changes
- Test changes
- CI-CD changes

## Purpose
To fix the following:
- The random password string for the SQL pool is strong and complex enough, so the deployment won't fail
- Passwords generated to later be saved on the KeyVault should not start with an hífen "-" or the az cli command should be able to accept it
- Be able to login to the right tenant where the deployment should be done and change de documentation
- Check AzDo pipeline existence prior creation and delete it to avoid failures if pipeline exists
- Fix ADF pipeline to use the latest REST API links
- CI and CD pipelines are running properly with no issues

## Does this introduce a breaking change? If yes, details on what can break

## Author pre-publish checklist

## Validation steps
- Don't define the SQL password on the env file - let the sample generate a random one
- Follow the new instructions to login into the right tenant
- Deploy sample
- Test end to end - running ADF pipelines and CICD pipelines
- Verify that everything is working with no errors and functionally
- Verify that there was no error thrown because of the complexity of the SQL password

## Issues Closed or Referenced
- Closes 
    - #856 
    - #823 
    - #685 
    - #824 
- References #822
